util/src/main/java/org/apollo/util/BufferUtil.java

```diff
@@ -65,10 +78,60 @@ public static int readUnsignedMedium(ByteBuffer buffer) {
 		return (buffer.getShort() & 0xFFFF) << 8 | buffer.get() & 0xFF;
 	}
 
+	/**
+	 * Reads a 24-bit medium integer from the specified {@link ByteBuffer}s current position and increases the buffers
+	 * position by 3.
+	 *
+	 * @param buffer The {@link ByteBuffer} to read from.
+	 * @return The read 24-bit medium integer.
+	 */
+	public static int readUnsignedMedium(ByteBuf buffer) {
+		return buffer.readUnsignedShort() << 8 | buffer.readUnsignedByte();
+	}
+
+	/**
+	 * Reads a jagex-string from the specified {@link ByteBuf}.
+	 *
+	 * @param buffer The buffer.
+	 * @return The string.
+	 */
+	public static String readJagexString(ByteBuf buffer) {
+		buffer.readByte();
+		return readString(buffer);
+	}
+
+
+	/**
+	 * Writes a string to the specified {@link ByteBuf}
+	 * @param buffer
+	 * @param str
+	 */
+	public static void writeString(ByteBuf buffer, String str) {
+		byte[] bytes = str.getBytes(CHARSET);
+		buffer.writeBytes(bytes);
+		buffer.writeByte(0);
+	}
+
+	/**
+	 * Writes a string to the specified {@link ByteBuf}
+	 * @param buffer
+	 * @param str
+	 */
+	public static void writeString(ByteBuffer buffer, String str) {
+		byte[] bytes = str.getBytes(CHARSET);
+		buffer.put(bytes);
+		buffer.put((byte) 0);
+	}
+
 	/**
 	 * The terminator of a string.
 	 */
-	public static final int STRING_TERMINATOR = 10;
+	public static final int STRING_TERMINATOR = 0;
+
+	/**
+	 * Charset that Jagex uses.
```

Comment:
RuneScape, not Jagex.

Reply:
Jagex calls it a jagString actually but whatever floats your boat.